elastic/apm-server
# sampling: add missing license check

## Motivation/summary

Tail-based sampling will be a licensed feature, add code to enforce it.

We also introduce support for running ephemeral Elasticsearch containers using testcontainers-go. This is used for running a Basic-licensed Elasticsearch without modifying the docker-compose services, which is destructive and may interfere with other tests or general docker-compose usage.

## Checklist

~- [ ] I have updated [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/master/CHANGELOG.asciidoc)~

I have considered changes for:
~- [ ] documentation~
~- [ ] metrics and monitoring (create issue for Kibana team to add metrics to visualizations, e.g. [Kibana#44001](https://github.com/elastic/kibana/issues/44001))~
~- [ ] telemetry~
~- [ ] Elasticsearch Service (https://cloud.elastic.co)~
~- [ ] Elastic Cloud Enterprise (https://www.elastic.co/products/ece)~
~- [ ] Elastic Cloud on Kubernetes (https://www.elastic.co/elastic-cloud-kubernetes)~

## How to test these changes

1. Run Elasticsearch with a trial license
2. Enable tail-based sampling -- it should work
3. Update Elasticsearch with a Basic license (https://www.elastic.co/guide/en/elasticsearch/reference/current/start-basic.html)
4. The server should stop working, and start logging licensing errors

## Related issues

https://github.com/elastic/apm-server/issues/4185
